### PR TITLE
Add expression filter to SKV Query

### LINKED ISSUE #60: Implement filter for query

### PR DESCRIPTION
**Pull request by jfunston**
closes #60

### CODE REVIEW

File: src/k2/module/k23si/Module.cpp [hunk 1]
```diff
@@ -268,6 +268,42 @@ dto::Key K23SIPartitionModule::_getContinuationToken(const IndexerIterator& it,
     };
 }
 
+// Makes the SKVRecord and applies the request's filter to it. If Status is not OK,
```
Comment (iafuture) [inline src/k2/module/k23si/Module.cpp:271]: why does bad status means that it should be returned?

File: src/k2/module/k23si/Module.cpp [hunk 2]
```diff
@@ -268,6 +268,42 @@ dto::Key K23SIPartitionModule::_getContinuationToken(const IndexerIterator& it,
...
+
+    dto::SKVRecord record(request.collectionName, versionIt->second);
+    record.storage = storage.share();
+    bool keep = false;
+    Status status = dto::K23SIStatus::OK("");
+
+    try {
+        keep = request.filterExpression.evaluate(record);
+    }
+    catch(dto::NoFieldFoundException&) {
```
Comment (iafuture) [inline src/k2/module/k23si/Module.cpp:291]: these two exceptions seem to be no-op. Why bother catching them?

File: src/k2/module/k23si/Module.cpp (hunk 1 above)
Comment (jfunston) [reply src/k2/module/k23si/Module.cpp:271]: If the status is bad from doQueryFilter, it should be returned to the user in the query response. That is what I mean here, not sure if that answers your question.

File: src/k2/module/k23si/Module.cpp (hunk 2 above)
Comment (jfunston) [reply src/k2/module/k23si/Module.cpp:291]: It needs to be caught so it doesn't propagate out. The keep = false is a no-op though and can be removed.

File: src/k2/module/k23si/Module.cpp (hunk 1 above)
Comment (iafuture) [reply src/k2/module/k23si/Module.cpp:271]: Oh i see. it is the status that is returned. I thought you're suggesting that the record should be returned if the status out of the filter is not OK.

**Review by iafuture — APPROVED** [review]

**Pull request merged**